sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3141,7 +3141,7 @@ class Analyzer(override val catalogManager: CatalogManager)
       // have been resolved.
       case p @ Project(projectList, child)
         if hasWindowFunction(projectList) && !p.expressions.exists(!_.resolved) =>
-        val (windowExpressions, regularExpressions) = extract(projectList)
+        val (windowExpressions, regularExpressions) = extract(projectList.toIndexedSeq)
```

Comment:
Just an idea: instead of fixing the internal code here and there, shall we simply fix all the public APIs that take `Seq`, to always call `toIndexedSeq` so that the `Stream` is materialized?

Reply:
>shall we simply fix all the public APIs that take Seq, to always call toIndexedSeq

I thought about that, and I think we should do that as well.

Although right now, AFAIK, the only way to pass lazy data structures to these loops is through the public API, there are other paths to this code (e.g., the parser) and some future innocuous-seeming change might end up passing streams to these loops. For example, somebody might add code that uses `Javaconverters`:
```
scala> jal
res7: java.util.ArrayList[Integer] = [2]

scala> jal.iterator.asScala.toSeq
res8: Seq[Integer] = Stream(2, ?)

scala> 
```